Documentation/policy.rst

```diff
@@ -0,0 +1,408 @@
+  for Layer 4 and Layer 7 rules.
+
+************
+Architecture
+************
+
+Rule Distribution
+=================
+
+Policy rules imported into the Cilium agent are not shared with other agents
+automatically. It is up to the control plane to distribute the rules to all
+compute nodes. If you are using Kubernetes, using the k8s_policy_ will
+automatically distribute the policy for you.
+
+.. note: Use of the KVstore to distribute security policies is on the roadmap
+         but has not been implemented yet.
+
+
+*********
+Base Type
+*********
+
+All policy rules share a common base type which specifies what endpoints the
+rule applies to and also carries common metadata to identify the rule.
+
+Each rule is split into an ingress section which contains the rules which must
+be applied at ingress and egress of all endpoint matching the endpoint
+selector.  Either ingress, egress, or both can be provided. If both ingress and
+egress are omitted, the rule has no effect.
+
+::
+
+        type Rule struct {
+                // EndpointSelector selects all endpoints which should be subject to
+                // this rule. Cannot be empty.
+                EndpointSelector EndpointSelector `json:"endpointSelector"`
+
+                // Ingress is a list of IngressRule which are enforced at ingress.
+                // If omitted or empty, this rule does not apply at ingress.
+                //
+                // +optional
+                Ingress []IngressRule `json:"ingress,omitempty"`
+
+                // Egress is a list of EgressRule which are enforced at egress.
+                // If omitted or empty, this rule does not apply at egress.
+                //
+                // +optional
+                Egress []EgressRule `json:"egress,omitempty"`
+
+                // Labels is a list of optional strings which can be used to
+                // re-identify the rule or to store metadata. It is possible to lookup
+                // or delete strings based on labels. Labels are not required to be
+                // unique, multiple rules can have overlapping or identical labels.
+                //
+                // +optional
+                Labels labels.LabelArray `json:"labels,omitempty"`
+
+                // Description is a free form string, it can be used by the creator of
+                // the rule to store human readable explanation of the purpose of this
+                // rule. Rules cannot be identified by comment.
+                //
+                // +optional
+                Description string `json:"description,omitempty"`
+        }
+
+
+*****************
+L3 (Labels Based)
+*****************
+
+The L3 policy specifies which endpoints can talk to each other. L3 policies can
+be specified using Labels_ or CIDR. For CIDR, refer to the policy_cidr_
+section. Label based L3 policy can only be specified on ingress using the
+``fromEndpoints`` field.  ``fromEndpoints`` is a list of endpoints identified
```

Comment:
If used in tandem with Kubernetes, you can use ``fromEndpoints``, a list of endpoints...